docs/src/snap/reference/certificates.md

```diff
@@ -0,0 +1,35 @@
+|------------------------------|--------------------------------------|--------------------------------------|-------------------------------------------|
+| `kubernetes-ca`               | General Kubernetes CA               | `/etc/kubernetes/pki/ca.crt`         | Signing all Kubernetes-related certificates |
+| `kubernetes-front-proxy-ca`   | CA for front-end proxy              | `/etc/kubernetes/pki/front-proxy-ca.crt` | Signing certificates for the front-proxy |
+| `client-ca`                   | CA for client certificates          | `/etc/kubernetes/pki/client-ca.crt` | Signing certificates for the client |
+
+
+#### 2. **Certificates**
+
+| **Common Name**                       | **Usage** | **Path on Disk**                                     | **Used For**                                                     | **Signed By**               |
+|--------------------------------------------|-----------|------------------------------------------------------|------------------------------------------------------------------|-----------------------------|
+| `kube-apiserver`                           | Server    | `/etc/kubernetes/pki/apiserver.crt`                  | Securing the API server endpoint                                 | `kubernetes-ca`             |
+| `kube-apiserver-kubelet-client`            | Client    | `/etc/kubernetes/pki/apiserver-kubelet-client.crt`   | API server communication with kubelets                           | `kubernetes-ca-client`      |
+| `kube-apiserver-etcd-client`               | Client    | `/etc/kubernetes/pki/apiserver-etcd-client.crt`      | API server communication with etcd                               | `kubernetes-ca-client`      |
+| `front-proxy-client`                       | Client    | `/etc/kubernetes/pki/front-proxy-client.crt`         | API server communication with the front-proxy                    | `kubernetes-front-proxy-ca` |
+| `kube-controller-manager`                  | Client    | `/etc/kubernetes/pki/controller-manager.crt`         | Communication between the controller manager and the API server  | `kubernetes-ca-client`      |
+| `kube-scheduler`                           | Client    | `/etc/kubernetes/pki/scheduler.crt`                  | Communication between the scheduler and the API server           | `kubernetes-ca-client`      |
+| `kube-proxy`                               | Client    | `/etc/kubernetes/pki/proxy.crt`                      | Communication between kube-proxy and the API server              | `kubernetes-ca-client`      |
+| `system:node:$hostname`                    | Client    | `/etc/kubernetes/pki/kubelet-client.crt`             | Authentication of kubelets to the API server                     | `kubernetes-ca-client`      |
+| `k8s-dqlite`             | Client    | `/var/snap/k8s/common/var/lib/k8s-dqlite/cluster.crt`             | Communication between k8s-dqlite nodes and API server | `self-signed`      |
```

Comment:
```suggestion
| `k8s`             | Client    | `/var/snap/k8s/common/var/lib/k8s-dqlite/cluster.crt`             | Communication between k8s-dqlite nodes and API server | `self-signed`      |
```

As per https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8sd/pki/k8sdqlite.go#L67 I think this is `k8s`.

Reply:
Again, we have two sources of truth here: https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8sd/pki/k8sdqlite.go#L21. I think we should be consistent and update the codebase as well. Another PR would be best, as this one is out of scope.

Reply:
I guess here the `k8s-dqlite` CN makes more sense - `k8s` is not very verbose and could maybe be confusing, WDYT?

Reply:
Nevermind, let's be consistent. In the comment above I said we go with whatever our codebase is doing. So `k8s` it is.

Reply:
We should have kube-apiserver-kubelet-client (which is standard as mateo mentioned) and k8s-dqlite  as the CN. The code can be changed in the next PR.

Reply:
As said, this change is not to be taken lightly and would need some extra handling for backward-compatibility.

We should rather document what we have now and update the docs along with the code change once we do them. I would not consider this super high prio right now (its not like we would fail any conformance with that - it's just an internal typo)